subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -494,4 +496,26 @@ static int backend_init(const struct device *instance)
 			 CONFIG_IPC_SERVICE_REG_BACKEND_PRIORITY,			\
 			 &backend_ops);
 
-DT_INST_FOREACH_STATUS_OKAY(DEFINE_BACKEND_DEVICE)
+DT_INST_FOREACH_STATUS_OKAY(BACKEND_DEVICE_DEFINE)
+
+#define BACKEND_CONFIG_DEFINE(i) BACKEND_CONFIG_POPULATE(i),
+
+static int shared_memory_prepare(const struct device *arg)
+{
+	const struct backend_config_t *backend_config;
+	const struct backend_config_t backend_configs[] = {
+		DT_INST_FOREACH_STATUS_OKAY(BACKEND_CONFIG_DEFINE)
+	};
+
+	for (backend_config = backend_configs;
+	     backend_config < backend_configs + ARRAY_SIZE(backend_configs);
+	     backend_config++) {
```

Comment:
This is unnecessarily complicated. `DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)` gives you the number of instances, and then you can simply get address and size with `DT_REG_{SIZE,ADDR}(DT_INST_PHANDLE(i, memory_region))`